tools/bumper/cnaoRepoCommands.go

```diff
@@ -0,0 +1,168 @@
+func openGitRepo(repoPath string) (*gitRepo, error) {
+	repo, err := git.PlainOpen(repoPath)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to open existing repo in path %s", repoPath)
+	}
+
+	return &gitRepo{
+		repo:     repo,
+		localDir: repoPath,
+	}, nil
+}
+
+func (cnaoRepoOps *gitCnaoRepo) getComponentsConfig(relativeConfigPath string) (componentsConfig, error) {
+	configPath := filepath.Join(cnaoRepoOps.gitRepo.localDir, relativeConfigPath)
+	return parseComponentsYaml(configPath)
+}
+
+func (cnaoRepoOps *gitCnaoRepo) updateComponentsConfig(relativeConfigPath string, componentsConfig componentsConfig) error {
+	configPath := filepath.Join(cnaoRepoOps.gitRepo.localDir, relativeConfigPath)
+	return updateComponentsYaml(configPath, componentsConfig)
+}
+
+func (cnaoRepoOps *gitCnaoRepo) bumpComponent(componentName string) error {
+	logger.Printf("Running bump-%s script", componentName)
+	cmd := exec.Command("make", "-C", cnaoRepoOps.gitRepo.localDir, fmt.Sprintf("bump-%s", componentName))
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err := cmd.Run()
+	if err != nil {
+		return errors.Wrapf(err, "Failed to run bump script, \nStdout:\n%s\nStderr:\n%s", cmd.Stdout, cmd.Stderr)
+	}
+	return nil
+}
+
+func (cnaoRepoOps *gitCnaoRepo) isComponentBumpNeeded(currentReleaseVersion, latestReleaseVersion, updatePolicy, proposedPrTitle string) (bool, error) {
+	logger.Printf("currentReleaseVersion: %s, latestReleaseVersion: %s, updatePolicy: %s\n", currentReleaseVersion, latestReleaseVersion, updatePolicy)
+
+	if updatePolicy == updatePolicyStatic {
+		logger.Printf("updatePolicy is static. avoiding auto bump")
+		return false, nil
+	}
+
+	// check if PR not already opened
+	isAlreadyOpened, err := cnaoRepoOps.isPrAlreadyOpened(cnaoRepoOps.getCnaoRepoOwnerFromUrl(), cnaoRepoOps.getCnaoRepoNameFromUrl(), proposedPrTitle)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to check if PR already open")
+	}
+	if isAlreadyOpened {
+		logger.Printf("Bump PR for the latest version already exist. Aborting auto bump")
+		return false, nil
+	}
+
+	// if one of the tags is in vtag format (e.g 0.39.0-32-g1fcbe815), and not equal, then always bump
+	if isVtagFormat(currentReleaseVersion) || isVtagFormat(latestReleaseVersion) {
+		return currentReleaseVersion == latestReleaseVersion, nil
+	}
+
+	currentVersion, err := canonicalizeVersion(currentReleaseVersion)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to digest current Version %s to semver", currentVersion)
+	}
+	latestVersion, err := canonicalizeVersion(latestReleaseVersion)
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to digest latest Version %s to semver", latestVersion)
+	}
+
+	return currentVersion.LessThan(*latestVersion), nil
+}
+
+// check if there is an already open PR with the same title in the repo.
+func (cnaoRepoOps *gitCnaoRepo) isPrAlreadyOpened(owner, repo, proposedPrTitle string) (bool, error) {
```

Comment:
owner and repo are going to be the ones at CNAO no need to pass them

Reply:
Done